app/src/main/java/com/infomaniak/mail/utils/LocalStorageUtils.kt

```diff
@@ -104,20 +120,39 @@ object LocalStorageUtils {
         }
     }
 
-    fun deleteAttachmentsUploadsDirIfEmpty(
+    fun deleteAttachmentDir(
+        context: Context,
+        draftLocalUuid: String,
+        attachmentLocalUuid: String,
+        userId: Int = AccountUtils.currentUserId,
+        mailboxId: Int = AccountUtils.currentMailboxId,
+    ) {
+
+        val attachmentDir = getAttachmentUploadDir(context, draftLocalUuid, attachmentLocalUuid, userId, mailboxId).also {
+            if (!it.exists()) return
+        }
+
+        // Only delete a directory if it's empty
+        if (attachmentDir.hasNoChildren) attachmentDir.delete()
+
+        deleteDraftDir(context, draftLocalUuid, userId, mailboxId)
+    }
+
+    fun deleteDraftDir(
         context: Context,
         draftLocalUuid: String,
         userId: Int = AccountUtils.currentUserId,
         mailboxId: Int = AccountUtils.currentMailboxId,
     ) {
-        val attachmentsDir = getAttachmentsUploadDir(context, draftLocalUuid, userId, mailboxId).also {
+        val draftDir = getDraftUploadDir(context, draftLocalUuid, userId, mailboxId).also {
             if (!it.exists()) return
         }
-        val mailboxDir = attachmentsDir.parentFile ?: return
+        val mailboxDir = draftDir.parentFile ?: return
         val userDir = mailboxDir.parentFile ?: return
         val attachmentsRootDir = userDir.parentFile ?: return
 
-        if (attachmentsDir.hasNoChildren) attachmentsDir.delete()
+        // Only delete a directory if it's empty
+        if (draftDir.hasNoChildren) draftDir.delete()
```

Comment:
we could return here, because all the other check are useless if we don't delete the draft dir